streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -526,15 +555,29 @@ public GroupAssignment assign(final Cluster metadata, final GroupSubscription gr
 
         // ---------------- Step Two ---------------- //
 
-        // assign tasks to clients
         final Map<UUID, ClientState> states = new HashMap<>();
         for (final Map.Entry<UUID, ClientMetadata> entry : clientMetadataMap.entrySet()) {
-            states.put(entry.getKey(), entry.getValue().state);
+            final ClientState state = entry.getValue().state;
+            states.put(entry.getKey(), state);
+
+            // Convert ownedPartitions to activeTasks if necessary
```

Comment:
If the subscription came from a client following the cooperative protocol, it will have embedded `ownedPartitions` instead of activeTasks